modules/core/src/smithy4s/internals/DocumentEncoderSchemaVisitor.scala

```diff
@@ -104,10 +104,10 @@ class DocumentEncoderSchemaVisitor(
         case HTTP_DATE => ts => DString(ts.format(HTTP_DATE))
         case EPOCH_SECONDS =>
           ts =>
-            val epochSecondsWithNanos =
-              BigDecimal(ts.epochSecond) + (BigDecimal(ts.nano) * BigDecimal(10)
-                .pow(-9))
-            DNumber(epochSecondsWithNanos)
+            val bd = (BigDecimal(ts.epochSecond) + (BigDecimal(
+              ts.nano
+            ) * BigDecimal(10).pow(-9))).underlying.stripTrailingZeros
```

Comment:
potential issue: this is different from the current behavior in the [regular encoders](https://github.com/disneystreaming/smithy4s/blob/e369dd40850690cb38efd385c1cb16ed3d09a192/modules/json/src/smithy4s/json/internals/SchemaVisitorJCodec.scala#L396), because it strips all the trailing zeros, rather than setting the scale to 9 at all times. Based on this [comment](https://github.com/disneystreaming/smithy4s/issues/1622#issuecomment-2500383589) jsoniter will respect the scales set on the `BigDecimal`, so striping all trailing zeros seem like a more general solution, rather than doing that only if `nano == 0`. 

Please let me know what you think, it definitely has to be consistent across both visitors, so I will adjust once this discussion is resolved.

Reply:
I'm happy with stripping trailing zeros, as long as the adjustment on the jsoniter-based codecs is not too complicated.

Reply:
Perhaps I missed something, but to me it looks like it's as straighforward as the `Document` fix.